app/components/rdf-input-fields/geboorte-datum-input.js

```diff
@@ -0,0 +1,108 @@
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { guidFor } from '@ember/object/internals';
+import InputFieldComponent from '@lblod/ember-submission-form-fields/components/rdf-input-fields/input-field';
+import { triplesForPath } from '@lblod/submission-form-helpers';
+import { replaceSingleFormValue } from 'frontend-lmb/utils/replaceSingleFormValue';
+import moment from 'moment';
+import { NamedNode } from 'rdflib';
+
+export default class RDFGeboorteDatumInput extends InputFieldComponent {
+  inputId = 'birthdate-' + guidFor(this);
+
+  @service store;
+  @tracked geboortedatum;
+
+  minDate;
+  maxDate;
+
+  constructor() {
+    super(...arguments);
+
+    const now = new Date();
+    const hundredYearsAgo = new Date(
+      now.getFullYear() - 100,
+      now.getMonth(),
+      now.getDate()
+    );
+    const eighteenYearsAgo = new Date(
+      now.getFullYear() - 18,
+      now.getMonth(),
+      now.getDate()
+    );
+
+    this.minDate = hundredYearsAgo;
+    this.maxDate = eighteenYearsAgo;
+
+    this.loadProvidedValue();
+  }
+
+  async loadProvidedValue() {
+    const matches = triplesForPath(this.storeOptions);
+    if (matches.values.length > 0) {
+      const queryResult = await this.store.query('geboorte', {
+        filter: { ':uri:': matches.values[0].value },
+      });
+      if (queryResult.length >= 1) {
+        const geboorte = queryResult.at(0);
+        this.geboortedatum = geboorte.datum;
+      }
+    }
+  }
+
+  @action
+  async updateValue(isoDate, date) {
+    this.geboortedatum = date;
+
+    this.updateValidations(date);
+
+    if (this.hasErrors) {
+      return;
+    }
+
+    const geboorte = await this.loadOrCreateGeboorte(isoDate, date);
+
+    replaceSingleFormValue(this.storeOptions, new NamedNode(geboorte.uri));
+
+    this.hasBeenFocused = true;
+  }
+
+  async loadOrCreateGeboorte(isoDate, date) {
+    let geboorte;
+    const queryResult = await this.store.query('geboorte', {
```

Comment:
this seems dangerous. It means that we reuse geboortes between person instances, which I wouldn't do.
Imagine someone wants to add additional info to the geboorte instance (like place of birth)

Suddenly you can't anymore.

I believe you will end up creating a new geboorte instance for every update (like we do for addresses).

We could get around this by making forms more complex, adding a nested object for geboorte (and address), but don't do this yet, it will take this story too far. I have another ticket for this.

Reply:
I'm still on the fence why we need to use a separate geboorte model, really curious what the thought was behind this. What you suggest would be a valid use case for implementing geboorte as a separate instance, still a meager one however. 

I saw it more like you want to group persons together based on their birthdate (for some unknown reason), but that indeed does not have much use. However this is also how it was done in legacy. 

But I can certainly change this If you want.